src/main/java/ch/heigvd/amt/stack/ui/web/views/ranking/RankingView.java

```diff
@@ -0,0 +1,29 @@
+package ch.heigvd.amt.stack.ui.web.views.ranking;
+
+import ch.heigvd.amt.stack.application.GamificationFacade;
+import java.io.IOException;
+import javax.inject.Inject;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet(name = "RankingView", urlPatterns = "/rankings")
+public class RankingView extends HttpServlet {
+
+  @Inject
+  GamificationFacade gamificationFacade;
+
+  @Override
+  protected void doGet(
+      HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    var page = req.getParameter("page");
+    req.setAttribute("rankings", gamificationFacade.getCategoryRankings(
+        gamificationFacade.getUsername(
+            gamificationFacade.getCredential(req.getSession().getId()).toString()),
```

Comment:
`getCredential` exposes a `CredentialId`, which should not be known by the view layer / `HttpServlet`